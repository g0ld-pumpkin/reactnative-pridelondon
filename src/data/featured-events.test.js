// @flow
import { generateCMSFeaturedEvents, sampleOne } from "./__test-data";
import decodeFeaturedEvents from "./featured-events";

describe("FeaturedEvents", () => {
  describe("decoder", () => {
    it("correctly decodes valid CMSFeaturedEvents", () => {
      const data: mixed = sampleOne(generateCMSFeaturedEvents);

      const decoded = decodeFeaturedEvents("en-GB")(data);
      expect(decoded.ok).toEqual(true);
      if (decoded.ok) {
        expect(decoded.value).toMatchSnapshot();
      }
    });

    it("fails if a property is missing", () => {
      const data: mixed = {
        fields: {},
        sys: {
          id: "3O3SZPgYl2MUEWu2MoK2oi",
          contentType: {
            sys: {
              id: "featuredEvents"
            }
          }
        }
      };

      const decoded = decodeFeaturedEvents("en-GB")(data);
      expect(decoded.ok).toEqual(false);
    });
  });
});
