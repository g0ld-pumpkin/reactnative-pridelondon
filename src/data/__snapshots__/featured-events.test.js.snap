// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeaturedEvents decoder correctly decodes valid CMSFeaturedEvents 1`] = `
Object {
  "contentType": "featuredEvents",
  "fields": Object {
    "events": Array [
      Object {
        "sys": Object {
          "id": "Ii3rT8J7D7oRAsi73e2G",
        },
      },
      Object {
        "sys": Object {
          "id": "k9XA4bnQ5lc3pH7IRmnTf7wdX9FLFT",
        },
      },
      Object {
        "sys": Object {
          "id": "78",
        },
      },
      Object {
        "sys": Object {
          "id": "A",
        },
      },
      Object {
        "sys": Object {
          "id": "66xFJ9CdsFOL2f3P68Muj8G9",
        },
      },
      Object {
        "sys": Object {
          "id": "fJ7T9A223S7qRj",
        },
      },
      Object {
        "sys": Object {
          "id": "7oFWgkXAs8AsHxk1kF2xG1syba124B",
        },
      },
      Object {
        "sys": Object {
          "id": "V04M29qi81C6Hvn",
        },
      },
      Object {
        "sys": Object {
          "id": "Vg8771gCXu861",
        },
      },
    ],
    "title": "title",
  },
  "id": "hG",
  "locale": "en-GB",
  "revision": 1,
}
`;
